samples/nrf_compress/mcuboot_update/src/main.c

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/init.h>
+#include <zephyr/mgmt/mcumgr/mgmt/mgmt.h>
+#include <zephyr/mgmt/mcumgr/mgmt/callbacks.h>
+#include <zcbor_common.h>
+#include <zcbor_encode.h>
+
+#ifndef CONFIG_MCUMGR_GRP_IMG_FRUGAL_LIST
+#define ZCBOR_ENCODE_FLAG(zse, label, value)                                    \
+                (zcbor_tstr_put_lit(zse, label) && zcbor_bool_put(zse, value))
+#else
+/* In "frugal" lists flags are added to response only when they evaluate to true */
+/* Note that value is evaluated twice! */
+#define ZCBOR_ENCODE_FLAG(zse, label, value)                                    \
+                (!(value) ||                                                    \
+                 (zcbor_tstr_put_lit(zse, label) && zcbor_bool_put(zse, (value))))
+#endif
+
+static struct mgmt_callback img_slot_callback;
+
```

Comment:
comment in the code for explaining what is it doing would be helpful once someone will look at this in few months.